src/main/java/billgates/view/gui/ActionPanel.java

```diff
@@ -249,7 +256,7 @@ private void addEntry() {
         int result = JOptionPane.showConfirmDialog(null, myPanel,
                 "Information about the new entry", JOptionPane.OK_CANCEL_OPTION);
         if (result == JOptionPane.OK_OPTION) {
-            String date1 = dateField.getText();
+            String date1 = datePicker2.getText();
```

Comment:
This does not correspond to the date format we defined. Therefore, this will crash the add entry function.

You can use `datePicker2.getDate()` to get a local date and convert the local date to a `ZonedDateTime` like what the code did at line 261.